staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -403,6 +426,45 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+// HACK: In some contexts, like the controller-runtime library used by the Operator SDK, all the resources of the
+// client-go scheme are created / updated using the protobuf content type.
```

Comment:
The solution is going to probably be negotiation of serialization type in client-go, which we haven't implemented.

Add a sentence at the bottom.

"When a request for an unsupported serialization is returned, the server should reject it with a 406 and provide a list of supported content types.  client-go should then examine whether it can satisfy such a request by encoding the object with a different scheme.  This would require a KEP but is in keeping with content negotiation on GET / WATCH in Kube."

Reply:
I added the new sentence to the HACK comment as well as in the commit message.